object-inspector/src/components/ObjectInspectorComponent.tsx

```diff
@@ -1,22 +1,115 @@
-import React from "react";
-import { Text } from "@chakra-ui/react";
+import React, { useState } from "react";
+import { Text, Input } from "@chakra-ui/react";
 import { ObjectInspector as ObjectorInspectorDS } from "@devtools-ds/object-inspector";
 import { ReactAsset } from "@player-ui/react";
+import { Flow } from "@player-ui/types";
 import { ObjectInspectorAsset } from "../types";
 
+const FilterObjectInspector = (props: ObjectInspectorAsset) => {
+  const { data, id } = props;
+
+  const [filterCriteria, setFilterCriteria] = useState("");
+  const [resultData, setResultData] = useState(data);
+
+  const isObject = (value: any): boolean => {
```

Comment:
Can we move `isObject` and `getPathvalue` from the body of the `FilterObjectInspector` to the `ObjectInspectorComponent.tsx` scope (AKA move it to line 7)?
In React, whenever a component re-renders, all the code inside the component's function body runs again. 
When we define pure functions (functions that always produce the same output given the same input) inside a component, they are recreated on every render, even though their logic doesn't change. 
This can lead to unnecessary memory usage and potentially slower performance, especially for large applications or components that re-render frequently.
Moving pure functions outside the component creates them once when the module loads rather than on every render. This can improve performance by reducing the amount of work React needs to do during re-renders. It also makes our code cleaner and easier to test, as pure functions are generally easier to reason about and isolate in tests than functions tied to a component's lifecycle.